### ISSUE TITLE
False positive for underscore argument name (`__`) with `reportMissingParameterType`

### ISSUE THREAD

**Issue by hilorioze** [microsoft/pyright#8884]
I'm getting false-positive error for an argument name with more than one underscore: `error: Type annotation is missing for parameter ‘__’ (reportMissingParameterType)`.
With single underscore it feels great.

Code sample in [pyright playground](https://pyright-play.net/?pyrightVersion=1.1.378&pythonVersion=3.12&reportMissingParameterType=true&code=CYUwZgBA9g1gFAfQJQQLQD4IDkoDsQBcEEAxBAAYBOIADlJQC4CyAlgM5su4DmACgIaV%2BAWxAMQlACoBPGiAC8DSgFcQ5CAHcWDABYROPADYgIy3KEpsAxvRNgQIQ2wjdq-BgChixag2WVcbDwQDw9QSAlKekRkNEwcfCJSCAAjZQYIXCgMrV0IfghgKGUU41NzCWtbL29ff0CEkLDwUzYQBCt%2BNrgAKgQAGggevtiMIMSanzF68ZCgA)

```python
def ok(_) -> None:  # `reportMissingParameterType=true` with single underscore feels great
    return None

def error(__) -> None:  # but not with a double underscore
    return None

def use_case(*_, **__) -> None:
    return None

```

Pyright version: `1.1.378`

**Comment by erictraut** [comment]
Pyright is working as intended here. A symbol with a single underscore in Python is commonly used to indicate "this isn't used", so it is specifically exempted from the `reportMissingParameterType` check. Other symbol names, including `__` are not exempted.

The `reportMissingParameterType` check is a strict-mode check, so it is somewhat opinionated. You can disable it if you don't like how it works. A linter like ruff might offer you a similar check with added configurability if that's what you're looking for.
